### PR TITLE
[wasm] reduce eval in production code

### LINKED ISSUE #61287: [wasm] avoid eval, design better API for InvokeJS
After change https://github.com/dotnet/runtime/pull/61212 Rollup started producing warning about eval.
```
(!) Use of eval is strongly discouraged
  https://rollupjs.org/guide/en/#avoiding-eval
  method-calls.ts
```

We could suppress it globally, but it's not ideal.
```
{
  onwarn(warning, warn) {
    if (warning.code === 'EVAL') return
    warn(warning)
  }
}
```

As @lewing  [suggested](https://github.com/dotnet/runtime/issues/61192), we should consider deprecating the method, since it's only internal API.

### PR DESCRIPTION
**Pull request by pavelsavara**
Fixes https://github.com/dotnet/runtime/issues/61287

### CODE REVIEW

**Review by maraf — APPROVED** [review]

**Comment by pavelsavara** [comment]
cc @kg

**Review by kg — APPROVED** [review]

**Comment by pavelsavara** [comment]
Sorry guys, I realized that there is still usage in the benchmark.
That also forced me to fix JSExport from nested class.

**Review by maraf — APPROVED** [review]

File: src/mono/sample/wasm/browser-bench/Program.cs [hunk 1]
```diff
@@ -72,6 +75,13 @@ public static void SetTasks(string taskNames)
             instance.tasks = tasksList;
         }
 
+        [JSExport]
+        [MethodImpl(MethodImplOptions.NoInlining)]
```
[the remark this answers is not in the document]
Comment (pavelsavara) [reply src/mono/sample/wasm/browser-bench/Program.cs:79]: Probably not but I'm not confident about it yet. There are many places like this. Let's do the research and solve them all at once.

**Comment by lewing** [comment]
https://radekdoulik.github.io/WasmPerformanceMeasurements/?startDate=2022-07-11T19%3A31%3A11.000Z&endDate=2022-07-11T21%3A06%3A41.000Z&tasks=%2CAppStart&flavors=2

**Comment by kg** [comment]
> https://radekdoulik.github.io/WasmPerformanceMeasurements/?startDate=2022-07-11T19%3A31%3A11.000Z&endDate=2022-07-11T21%3A06%3A41.000Z&tasks=%2CAppStart&flavors=2

The time increase in this doesn't make any sense given the change in question, seems like variance?

**Comment by lewing** [comment]
https://github.com/dotnet/runtime/compare/cae954add0f01c85de7fef8cc41c2a11d1c920e1...4a2ebf204801dc099d4e58dcdaf715dcecedcfda is the only commit in the diff and it changes the test mechanism

**Comment by pavelsavara** [comment]
Yes, I had to rewrite the benchmark to stop using eval code path and so I introduced dependency on JSExport. The implementation at the time was triggering static constructors of the containing class during runtime init. That's now fixed in https://github.com/dotnet/runtime/pull/77293

**Comment by radekdoulik** [comment]
> https://radekdoulik.github.io/WasmPerformanceMeasurements/?startDate=2022-07-11T19%3A31%3A11.000Z&endDate=2022-07-11T21%3A06%3A41.000Z&tasks=%2CAppStart&flavors=2

yup, it is https://github.com/dotnet/runtime/issues/75598 and Pavel fixed it in https://github.com/dotnet/runtime/pull/77293. the static constructors were doing a lot of work, which was not needed during startup

**Pull request merged**